main.go

```diff
@@ -92,7 +91,14 @@ func run() int {
 	level.Info(logger).Log("msg", "Loaded config file")
 
 	// Infer or set Blackbox exporter externalURL
-	beURL, err := computeExternalURL(*externalURL, *listenAddress)
+	listenAddrs := toolkitFlags.WebListenAddresses
+	if *externalURL == "" && *toolkitFlags.WebSystemdSocket {
```

Comment:
Shouldn't this be `*externalURL != ""`

Reply:
I believe it is correct, and I tested it. This if statement intentionally throws an error if an external URL is not explicitly provided when using a systemd socket listener. This is because the external URL can never be inferred from the systemd socket.

Reply:
But, what if you don't care about the external URL? It's an optional setting in the normal case. Wouldn't this for the user to set an external URL in systemd mode? I don't think that's desired.

Reply:
External URL is an optional setting because it infers it from the port number or hostname provided when no external URL is provided. The default from the optional flag is an empty string, but then the function called on it later finds the real fallback value.

If you're saying the external URL is not actually needed for the exporter to function, then I think that goes past the scope of this pull.